grammars/javascript.cson

```diff
@@ -1121,67 +1094,117 @@
         ]
       }
     ]
-  'methods':
+  'arrow_functions':
     'patterns': [
       {
-        'name': 'meta.method.js'
-        'comment': 'match regular function like: function myFunc(arg) { … }'
-        'begin': '\\b((?!(?:break|case|catch|continue|do|else|finally|for|function|if|export|import|package|return|switch|throw|try|while|with)[\\s\\(])(?:[a-zA-Z_$][a-zA-Z_$0-9]*))\\s*(\\()(?=(?:[^\\(\\)]*)?\\)\\s*\\{)'
+        # (a,b) =>
+        'begin': '(\\()(?=(?:[^\\(\\)]*)?\\)\\s*=>)'
         'beginCaptures':
           '1':
-            'name': 'entity.name.function.js'
-          '2':
             'name': 'punctuation.definition.parameters.begin.js'
-        'end': '(\\))'
+        'end': '(\\))\\s*(=>)'
         'endCaptures':
           '1':
             'name': 'punctuation.definition.parameters.end.js'
+          '2':
+            'name': 'storage.type.function.arrow.js'
         'patterns': [
           {
-            'include': '#function-params'
-          }
-          {
-            'include': '#strings'
+            'include': '#function_params'
           }
         ]
       }
     ]
-  'function-params':
+  'function_params':
     'patterns': [
       {
-        'include': '#numbers'
+        'begin': '(\\()'
+        'beginCaptures':
+          '1':
+            'name': 'punctuation.definition.parameters.begin.bracket.round.js'
+        'end': '(\\))'
+        'endCaptures':
+          '1':
+            'name': 'punctuation.definition.parameters.end.bracket.round.js'
+        'name': 'meta.parameters.js'
+        'patterns': [
+          {
+            'match': '(\\.\\.\\.)([a-zA-Z_$][a-zA-Z_$0-9]*)'
+            'captures':
+              '1':
+                'name': 'keyword.operator.spread.js'
+              '2':
+                'name': 'variable.parameter.rest.function.js'
+          }
+          {
+            'include': '$self'
+          }
+          {
+            'match': '[a-zA-Z_$][a-zA-Z_$0-9]*'
```

Comment:
e.g. `this` in `function callSomething(thing = this.something()) {}` should be matched first.
I am not sure what else can be inside function parameters, so put the pattern below `$self`